tensorboard/webapp/notification_center/_data_source/notification_center_data_source_test.ts

```diff
@@ -84,29 +84,40 @@ describe('TBNotificationCenterDataSource test', () => {
     expect(errorSpy).toHaveBeenCalledWith(httpErrorResponse);
   });
 
-  it(
-    'clips dateInMs to max today to prevent issue where notification never ' +
-      'gets dismissed',
-    () => {
-      jasmine.clock().mockDate(new Date('2000-01-31 07:00'));
-      const resultSpy = jasmine.createSpy();
-      dataSource.fetchNotifications().subscribe(resultSpy);
-      const req = httpMock.expectOne('/data/notifications');
-      req.flush(
-        buildNotificationResponse([
-          buildNotification({
-            dateInMs: new Date('2000-01-31 23:59').getTime(),
-          }),
-        ])
-      );
-
-      expect(resultSpy).toHaveBeenCalledOnceWith(
-        buildNotificationResponse([
-          buildNotification({
-            dateInMs: new Date('2000-01-31 12:00 AM').getTime(),
-          }),
-        ])
-      );
-    }
-  );
+  it('filters notifications from the future', () => {
+    jasmine.clock().mockDate(new Date('2000-01-31 07:00'));
+    const resultSpy = jasmine.createSpy();
+    dataSource.fetchNotifications().subscribe(resultSpy);
+    const req = httpMock.expectOne('/data/notifications');
+    req.flush(
+      buildNotificationResponse([
+        buildNotification({
+          dateInMs: new Date('2000-01-31 07:00').getTime(),
+        }),
+        buildNotification({
+          dateInMs: new Date('2000-01-31 06:59').getTime(),
+        }),
+        buildNotification({
+          dateInMs: new Date('2000-01-31 07:01').getTime(),
+        }),
+        buildNotification({
+          dateInMs: new Date('2000-01-31 06:58').getTime(),
+        }),
```

Comment:
just a note that I don't think the notifications will be sort in order automatically.
For example, here the order is (7:00, 6:59, 6:58) then we should them in that order. (Let's exclude future notification)
but if that order is (6:59, 7:00, 6:58) we then show 6:59 as the most top notification not the most recent 7:00 one.

Is that correct? If so, can we address this in line 40-43 of data_surce.ts? (No need to be in the same PR)

Reply:
I did think about this: I think it's fine (for now) not to sort.  It keeps the code simpler and I think we can rely on whoever edits the notifications to send them in the order they want. Who knows - maybe there is a legitimate reason to show them out of order.